lib/src/components/NxAccordion/__tests__/NxAccordion.test.tsx

```diff
@@ -5,313 +5,210 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { mount } from 'enzyme';
-import { faChevronCircleDown, faChevronCircleUp } from '@fortawesome/free-solid-svg-icons';
-import { act } from 'react-dom/test-utils';
+import userEvent from '@testing-library/user-event';
 
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
-import NxAccordion, { Props } from '../NxAccordion';
-import NxButton from '../../NxButton/NxButton';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
 
-function createClickEvent() {
-  return new MouseEvent('click', { bubbles: true });
-}
+import NxAccordion from '../NxAccordion';
+import NxButton from '../../NxButton/NxButton';
+import { render } from '@testing-library/react';
 
 describe('NxAccordion', function() {
-  const getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxAccordion, {}),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxAccordion, {});
+  const quickRender = rtlRender(NxAccordion, {});
+  const renderEl = rtlRenderElement(NxAccordion, {});
 
   it('renders a <details> element with the provided props', function() {
-    const component = getShallowComponent({ id: 'foo', open: true, title: 'test title' });
-
-    expect(component).toMatchSelector('details#foo');
-    expect(component).toHaveProp('open', true);
-    expect(component).toHaveProp('title', 'test title');
+    expect(quickRender({ id: 'foo' }).getByRole('group')).toHaveAttribute('id', 'foo');
```

Comment:
We don't just want to test that some child is a `group` with the expected id, we want to test that the _top level element_ is a `group` with the expected id.

Reply:
> We don't just want to test that some child is a `group` with the expected id, we want to test that the _top level element_ is a `group` with the expected id.

Like this? 2caf1192854b872e28771e153d04634e4cf24e1e

Reply:
> Like this? https://github.com/sonatype/sonatype-react-shared-components/commit/2caf1192854b872e28771e153d04634e4cf24e1e

That's one way.  I would have just used `renderEl` instead of `quickRender` though.  I mispoke slightly above as (related to some of my other comments) the `group` role isn't that important here and I'd rather we check the element tag name instead.  So in summary, I think this test should use `renderEl` to get the top-level element, and then test that that element is a `details` and that its id is `foo`.

Reply:
> > Like this? [2caf119](https://github.com/sonatype/sonatype-react-shared-components/commit/2caf1192854b872e28771e153d04634e4cf24e1e)
> 
> That's one way. I would have just used `renderEl` instead of `quickRender` though. I mispoke slightly above as (related to some of my other comments) the `group` role isn't that important here and I'd rather we check the element tag name instead. So in summary, I think this test should use `renderEl` to get the top-level element, and then test that that element is a `details` and that its id is `foo`.

Updated here: 24a336d3c82d2224871268ce37740dc8275c6565